sdk/containerregistry/azure-containerregistry/azure/containerregistry/aio/_async_container_registry_client.py

```diff
@@ -49,28 +58,111 @@ async def delete_repository(self, repository: str, **kwargs: Dict[str, Any]) ->
         return DeletedRepositoryResult._from_generated(result)  # pylint: disable=protected-access
 
     @distributed_trace
-    def list_repositories(self, **kwargs) -> AsyncItemPaged[str]:
+    def list_repositories(self, **kwargs: Dict[str, Any]) -> AsyncItemPaged[str]:
         """List all repositories
 
         :keyword last: Query parameter for the last item in the previous call. Ensuing
             call will return values after last lexicallyy
         :type last: str
         :keyword max: Maximum number of repositories to return
         :type max: int
-        :keyword results_per_page: Numer of repositories to return in a single page
+        :keyword results_per_page: Numer of repositories to return per page
         :type results_per_page: int
         :return: ItemPaged[str]
         :rtype: :class:`~azure.core.async_paging.AsyncItemPaged`
         :raises: :class:`~azure.core.exceptions.ResourceNotFoundError`
         """
+        n = kwargs.pop("results_per_page", None)
+        last = kwargs.pop("last", None)
 
-        return self._client.container_registry.get_repositories(
-            last=kwargs.pop("last", None), n=kwargs.pop("page_size", None), **kwargs
-        )
+        cls = kwargs.pop("cls", None)
+        error_map = {401: ClientAuthenticationError, 404: ResourceNotFoundError, 409: ResourceExistsError}
+        error_map.update(kwargs.pop("error_map", {}))
+        accept = "application/json"
+
+        def prepare_request(next_link=None):
```

Comment:
If this method is the same as the sync `prepare_request`, would it be possible to place this in a shared base class?

Reply:
It probably is, this all had to be copied from the generated code. I'll open an issue to get this done over the next sprint. Thanks for the rec. #17817